libs/lb/lb-rs/src/lib.rs

```diff
@@ -35,13 +35,40 @@ pub struct Lb {
     pub syncing: Arc<AtomicBool>,
 }
 
+impl Lb {
+    pub fn init_dummy(config: Config) -> LbResult<Self> {
+        let db = CoreDb::init(db_rs::Config {
+            path: Default::default(),
+            create_path: false,
+            create_db: false,
+            read_only: false,
+            no_io: true,
+            fs_locks: false,
+            fs_locks_block: false,
+            schema_name: Default::default(),
+        })
+        .map_err(|err| LbErrKind::Unexpected(format!("db rs creation failed: {:#?}", err)))?;
+
+        Ok(Self {
+            config: config.clone(),
+            keychain: Default::default(),
+            db: Arc::new(RwLock::new(db)),
+            docs: AsyncDocs::from(&config),
+            search: Default::default(),
+            client: Default::default(),
+            syncing: Default::default(),
+            events: Default::default(),
+        })
+    }
+}
+
 impl Lb {
     #[instrument(level = "info", skip_all, err(Debug))]
     pub async fn init(config: Config) -> LbResult<Self> {
         logging::init(&config)?;
 
         let db = CoreDb::init(db_rs::Config::in_folder(&config.writeable_path))
-            .map_err(|err| LbErrKind::Unexpected(format!("{:#?}", err)))?;
+            .map_err(|err| LbErrKind::Unexpected(format!("db rs creation failed: {:#?}", err)))?;
```

Comment:
when you rebase you'll have access to the `.unexpected()` implementation from the extension trait in `errors.rs`, I suggest changing this to that.